modules/table/src/main/java/org/apache/ignite/internal/table/distributed/schema/CatalogVersionSufficiency.java

```diff
@@ -35,6 +35,6 @@ private CatalogVersionSufficiency() {
      * @return {@code true} iff the local Catalog version is sufficient.
      */
     public static boolean isMetadataAvailableFor(int requiredCatalogVersion, CatalogService catalogService) {
-        return requiredCatalogVersion <= catalogService.latestCatalogVersion();
+        return catalogService.catalogReadyFuture(requiredCatalogVersion).isDone();
```

Comment:
Could you please make an analogous change in `IncomingSnapshotCopier#metadataIsSufficientlyComplete()`?

Reply:
Probably it makes sense to just reuse the method there

Reply:
Fixed.